Packs/CortexXDR/Scripts/XDRIncidentWidgetBase/XDRIncidentWidgetBase.py

```diff
@@ -0,0 +1,60 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def main():
+    try:
+        res_type = demisto.getArg('reultType')
+        query_type = demisto.getArg('queryType')
+        to_date = demisto.getArg('to')
+        from_date = demisto.getArg('from')
+        list_name = f'xdrIncidents_{query_type}'
+
+        # parse time arguments
+        date_pattern = re.compile('\d{4}[-/]\d{2}[-/]\d{2}T\d{2}:\d{2}:\d{2}')
+
+        if to_date and to_date != '0001-01-01T00:00:00Z':
+            result = date_pattern.findall(to_date)[0]
+            to_date = datetime.strptime(result, '%Y-%m-%dT%H:%M:%S')
+        else:
+            to_date = datetime.max
+
+        if from_date and from_date != '0001-01-01T00:00:00Z':
+            result = date_pattern.findall(from_date)[0]
+            from_date = datetime.strptime(result, '%Y-%m-%dT%H:%M:%S')
+        else:
+            from_date = datetime.min
+
+        list_res = demisto.executeCommand("getList", {"listName": list_name})
+        if isError(list_res):
+            return_error(f'Error occurred while trying to get the list {list_name}: {get_error(list_res)}')
+
+        incidents = json.loads(list_res[0]["Contents"])
```

Comment:
It's better to have error handling even though you check the result with `is_error`